### PR TITLE
Redistribute nugetizer vsix and rework chocolatey install script to b…

### PR DESCRIPTION
**Pull request by bbowman**
…e less failure prone for systems that don't have latest visual studios

### CODE REVIEW

**Comment by bbowman** [comment]
@kzu probably cares about how we do this. Also how we attribute the nugetizer stuff to him. I'd love a review on this.

File: tools/WinObjC.Tools/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿#NOTE: Please remove any commented lines to tidy up prior to releasing the package, including this one
 
 # stop on all errors
```
Comment (rajsesh-msft) [inline tools/WinObjC.Tools/tools/chocolateyinstall.ps1]: Comment is now invalid.

File: tools/WinObjC.Tools/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -17,17 +17,45 @@ Write-Host "Adding `'$toolsDir`' to the path and the current shell path"
...
+}
+Catch [Exception]
+{
+    Write-Warning "$_"
+}
+
+Try
+{
+    Write-Host "Installing Objective-C Visual Studio 2017 Extension ... "
+    Install-ChocolateyVsixPackage -PackageName "objc-syntax-highlighting" -VsixUrl "file:///$toolsDir/objc-language-services-VS2017.vsix" -VsVersion 14
```
Comment (memontic-ms) [inline tools/WinObjC.Tools/tools/chocolateyinstall.ps1]: >14 [](start = 149, length = 2)

Intended 15?

File: tools/WinObjC.Tools/WinObjC.Tools.targets [hunk 3]
```diff
@@ -64,7 +64,7 @@
     <PropertyGroup>
       <PackageTimestamp Condition="'$(PackageTimestamp)' == ''">$([System.DateTime]::Now.ToString(yyyyMMddHHmmss))</PackageTimestamp>
       <PackageVersion Condition="'$(GitBranch)' == 'master'">$(GitBaseVersionMajor).$(GitBaseVersionMinor).$(PackageTimestamp)</PackageVersion>
-      <PackageVersion Condition="'$(PackageVersion)' == ''">$(GitBaseVersionMajor).$(GitBaseVersionMinor).$([MSBuild]::Add('$(GitBaseVersionPatch)', '1'))-$(PackageTimestamp)pr</PackageVersion>
+      <PackageVersion Condition="'$(PackageVersion)' == ''">$(GitBaseVersionMajor).$(GitBaseVersionMinor).$([MSBuild]::Add('$(GitBaseVersionPatch)', '1'))-pr-$(PackageTimestamp)</PackageVersion>
```
Comment (memontic-ms) [inline tools/WinObjC.Tools/WinObjC.Tools.targets]: Why is this change needed?

File: tools/WinObjC.Tools/tools/chocolateyinstall.ps1 (hunk 2 above)
Comment (bbowman) [inline tools/WinObjC.Tools/tools/chocolateyinstall.ps1]: yes thanks

**Comment by bbowman** [comment]
@kzu We still need to support VS2015 consumers (until 2017 comes out and is validated .. which I know is soon). My plan until then was to manually redist that vsix since it still works in our scenarios. 

From the sound of it you are confident with the 2017 evergreen url not chaning soon? I don't want to break the install script unnecessarily.

File: tools/WinObjC.Tools/WinObjC.Tools.targets (hunk 3 above)
Comment (bbowman) [inline tools/WinObjC.Tools/WinObjC.Tools.targets]: chocolatey appears to have a slightly different SemVer parser that didn't like the other style of version string.

**Comment by kzu** [comment]
The legacy 2015 vsix generates projects that are not aligned with VS2017. 

Given we're only a couple weeks from the official release of VS2017, I wouldn't set the wrong expectation from the get go.

File: tools/WinObjC.Tools/tools/chocolateyinstall.ps1 [hunk 4]
```diff
@@ -1,9 +1,7 @@
 ﻿#NOTE: Please remove any commented lines to tidy up prior to releasing the package, including this one
 
-# stop on all errors
 $ErrorActionPreference = 'Stop';
 
-
 $packageName = 'WinObjC.tools' # arbitrary name for the package, used in messages
```
Comment (memontic-ms) [inline tools/WinObjC.Tools/tools/chocolateyinstall.ps1]: >t [](start = 24, length = 1)

nit: Capital T

**Comment by bbowman** [comment]
@kzu @memontic-ms @rajsesh-msft any more feedback here?

**Comment by memontic-ms** [comment]
:shipit:
